conda_build/inspect_pkg.py

```diff
@@ -87,23 +93,21 @@ def __str__(self):
 untracked_package = _untracked_package()
 
 
+@deprecated.argument("3.28.0", "4.0.0", "package", rename="subdir")
+@deprecated.argument("3.28.0", "4.0.0", "prepend")
+@deprecated.argument("3.28.0", "4.0.0", "minimal_hint")
 def check_install(
-    packages, platform=None, channel_urls=(), prepend=True, minimal_hint=False
-):
-    prefix = tempfile.mkdtemp("conda")
-    try:
-        specs = specs_from_args(packages)
-        index = get_index(
-            channel_urls=channel_urls, prepend=prepend, platform=platform, prefix=prefix
-        )
-        actions = install_actions(
-            prefix, index, specs, pinned=False, minimal_hint=minimal_hint
-        )
-        display_actions(actions, index)
-        return actions
-    finally:
-        rm_rf(prefix)
-    return None
+    packages: Iterable[str],
+    subdir: str | None = None,
+    channel_urls: Iterable[str] = (),
+) -> None:
+    with TemporaryDirectory() as prefix:
+        Solver(
+            prefix,
+            channel_urls,
+            [subdir],
+            specs_from_args(packages),
+        ).solve_for_transaction().print_transaction_summary()
```

Comment:
```suggestion
        ).solve_for_transaction(ignore_pinned=True).print_transaction_summary()
```

Do we need this here due to the `pinned=False` in the original block?

Reply:
Good catch